subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/logging/ClassMethodNameStackTraceSpec.java

```diff
@@ -19,19 +19,34 @@
 import org.gradle.api.specs.Spec;
 
 import javax.annotation.Nullable;
+import java.util.regex.Pattern;
 
 public class ClassMethodNameStackTraceSpec implements Spec<StackTraceElement> {
+    private static final Pattern ANONYMOUS_CLASS_NAME_SUFFIX = Pattern.compile("[\\d$]+");
     private final String className;
     private final String methodName;
 
-    ClassMethodNameStackTraceSpec(@Nullable String className, @Nullable String methodName) {
+    ClassMethodNameStackTraceSpec(String className, @Nullable String methodName) {
         this.className = className;
         this.methodName = methodName;
     }
 
     @Override
     public boolean isSatisfiedBy(StackTraceElement element) {
-        return (className == null || className.equals(element.getClassName()))
-                && (methodName == null || methodName.equals(element.getMethodName()));
+        return classNameMatch(element.getClassName()) && methodNameMatch(element.getMethodName());
+    }
+
+    private boolean methodNameMatch(String methodName) {
+        return this.methodName == null || this.methodName.equals(methodName);
+    }
+
+    private boolean classNameMatch(String targetClassName) {
+        if (!targetClassName.startsWith(className)) {
+            return false;
+        }
+
+        String suffix = targetClassName.substring(className.length());
```

Comment:
This is probably not very important, but a more efficient algorithm wouldn't allocate a substring and wouldn't use a regex, but do a simple index scan over the String.

E.g.
```
if (!cls.startsWith(expected)) {
  return false
}
if (cls.length() == expected.length()) {
  return true
}
char firstExtraChar = cls.charAt(expected.length())
if (firstExtraChar != "$") {
  return false
}
char secondExtraChar = cls.charAt(expected.length() +1)
if (!secondExtraChar.isDigit) {
  return false
}
return true
```

Reply:
Yeah thanks, I convinced myself that it's simpler and won't bring too much overhead.